nms/app/views/events/EventsTable.js

```diff
@@ -262,11 +262,13 @@ export default function EventsTable(props: EventTableProps) {
 
   useEffect(() => {
     if (props.inStartDate && props.inEndDate) {
+      props.inStartDate !== undefined && setStartDate(props.inStartDate);
+      props.inEndDate !== undefined && setEndDate(props.inEndDate);
       if (tableRef.current) {
         tableRef.current.onQueryChange();
```

Comment:
this behaviour had already been present before my commit so I didn't touch although @thmsschmitt  said to remove.

Reply:
I think the motivation for `tableRef.current.onQueryChange` here was to refetch if props.inStartDate or props.inEndDate change. Which as you noticed doesn't work because the query uses the state and not the props. We don't have to `onQueryChange` explicitly because  `setStartDate` and `setEndDate` do also call it.

I would suggest something like:

```
useEffect(() => {
  if (props.inStartDate) {
    setStartDate(props.inStartDate);
  }
}, [props.inStartDate, setStartDate]);

useEffect(() => {
  if (props.inEndDate) {
    setEndDate(props.inEndDate);
  }
}, [props.inEndDate, setEndDate]);
```